default.nix

```diff
@@ -1,30 +1,21 @@
-let
-  # Names nixpkgs pins (corrisponds to names of the nixpkgs/*.json files)
-  nixpkgs-names = ["release-18.08" "release-19.03" "release-19.09"];
-
-  # Name of the default nixpkgs-pin
-  default-nixpkgs-name = "release-19.09";
-
-  # These are from lib.attrset but we don't want to have to import a nixpkgs just to get them
-  nameValuePair = name: value: { inherit name value; };
-  genAttrs = names: f: builtins.listToAttrs (builtins.map (n: nameValuePair n (f n)) names);
-
-in {
+rec {
   # nixpkgsArgs is designed to be passed directly nixpkgs with something like:
   #   import <nixpkgs> (import (builtins.fetchTarball https://github.com/input-output-hk/haskell.nix/archive/master.tar.gz)).nixpkgsArgs
   nixpkgsArgs = {
     config   = import ./config.nix;
     overlays = import ./overlays;
   };
 
-  inherit default-nixpkgs-name;
-  
   # nixpkgs allows you to import haskell.nix with the default pinned nixpkgs
   #   (import (builtins.fetchTarball https://github.com/input-output-hk/haskell.nix/archive/master.tar.gz)).nixpkgs {}
-  nixpkgs = args: import ./nixpkgs (args // { nixpkgs-pin = default-nixpkgs-name; });
+  defaultNixpkgs = nixpkgs-pins."19.09";
   
   # nixpkgs-pins allows you to import haskell.nix with one of the pinned nixpkgs
   #   (import (builtins.fetchTarball https://github.com/input-output-hk/haskell.nix/archive/master.tar.gz)).nixpkgs-pins."release-19.03" {}
-  nixpkgs-pins = (genAttrs nixpkgs-names
-      (nixpkgs-name: args: import ./nixpkgs (args // { nixpkgs-pin = nixpkgs-name; })));
+  nixpkgs-pins = builtins.mapAttrs
+    (_: nixpkgs-name: import ./nixpkgs { nixpkgs-pin = nixpkgs-name; }) {
+      "18.09" = "release-18.09";
+      "19.03" = "release-19.03";
+      "19.09" = "release-19.09";
```

Comment:
Sorry, I wasn't trying to be opinionated about the names! I think it would be fine to just use `release-19.09` as the name, if you don't want this awkward mapping!

Reply:
I actually liked to have an excuse to use `mapAttrs` as it is builtins, `genAttrs` had to be copied or imported from nixpkgs.  I'm happy to switch it back if you think we should use `release-` though.